Tests/SwiftGodotMacrosTests/TextureLiteralMacroTests.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  TextureLiteralMacroTests.swift
+//  SwiftGodot
+//
+//  Created by Marquis Kurt on 6/11/23.
+//
+
+import SwiftSyntaxMacros
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+import SwiftGodotMacroLibrary
+
+final class TextureLiteralMacroTests: XCTestCase {
+    let testMacros: [String: Macro.Type] = [
+        "texture2DLiteral": Texture2DLiteralMacro.self
+    ]
+
+    func testMacroExpansion() {
+        assertMacroExpansion(
+            """
+            let spriteTexture = #texture2DLiteral("res://assets/icon.png")
+            """,
+            expandedSource: """
+            let spriteTexture = {
+                guard let texture: Texture2D = GD.load(path: "res://assets/icon.png") else {
+                    preconditionFailure(
+                        "Texture could not be loaded.",
+                        file: "TestModule/test.swift",
+                        line: 1)
```

Comment:
Just curious, is there a way to see this "Texture could not be loaded." text when the game hits it? Any fatalError / preconditionFailures I've tried just leads to a crash in godot, and even the crash doesn't seem to show the actual failure text.

Reply:
Not to my knowledge, but I could call a `GD.pushError` beforehand so that the error is present in the Godot console before the crash.

Reply:
Thanks for getting back to me! I just tried this and the error doesn't make it to the Godot console unfortunately. No worries either way, this is not a reason to change the PR, which is really cool overall! Thanks for sharing it.

Reply:
Ah, I guess it must happen so quickly that Godot doesn't even have the chance to report the error. Though I wonder if it appears in the stacktrace or logs.

Reply:
Appears this might require a flag of some kind:

> `message`. 
> A string to print in a playground or -Onone build. The default is an empty string.

https://developer.apple.com/documentation/swift/preconditionfailure(_:file:line:)

Reply:
A debug build should be `-Onone` already, I suspect it's dying before the logs are flushed. I'll open a separate issue to discuss this.

Reply:
Opened https://github.com/migueldeicaza/SwiftGodot/issues/132 to discuss.

Reply:
@PadraigK Perhaps you can try replacing my `GD.pushError` with Swift's default `print` and see if you get something in the terminal?